common/js/src/popup-window/in-popup-main-script.js

```diff
@@ -59,20 +64,32 @@ GSC.InPopupMainScript.showWindow = function() {
   chrome.app.window.current().show();
 };
 
+const popUpData = GSC.InPopupMainScript.getData();
+
 /**
  * Resolves the modal dialog, passing the specified result to the caller in the
  * parent window (i.e. in the App's background page) and closing the current
  * window.
  * @param {*} result
  */
 GSC.InPopupMainScript.resolveModalDialog = function(result) {
-  const callback = GSC.InPopupMainScript.getData()['resolveModalDialog'];
-  GSC.Logging.checkWithLogger(logger, callback);
   goog.log.fine(
       logger,
       'The modal dialog is resolved with the following result: ' +
           GSC.DebugDump.debugDump(result));
-  callback(result);
+  if (GSC.Packaging.MODE === GSC.Packaging.Mode.APP) {
+    const callback = GSC.InPopupMainScript.getData()['resolveModalDialog'];
```

Comment:
nit: Let's use the new `popUpData` variable here and in other places throughout the file. It'll save on extra calls, but more importantly make the code look less surprising.